driver/mounter_mock.go

```diff
@@ -0,0 +1,224 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: mounter.go
+
+// Package driver is a generated GoMock package.
+package driver
+
+import (
+	reflect "reflect"
+
+	gomock "github.com/golang/mock/gomock"
+	mount_utils "k8s.io/mount-utils"
+)
+
+// MockiAttachmentValidator is a mock of iAttachmentValidator interface.
+type MockiAttachmentValidator struct {
+	ctrl     *gomock.Controller
+	recorder *MockiAttachmentValidatorMockRecorder
+}
+
+// MockiAttachmentValidatorMockRecorder is the mock recorder for MockiAttachmentValidator.
+type MockiAttachmentValidatorMockRecorder struct {
+	mock *MockiAttachmentValidator
+}
+
+// NewMockiAttachmentValidator creates a new mock instance.
+func NewMockiAttachmentValidator(ctrl *gomock.Controller) *MockiAttachmentValidator {
+	mock := &MockiAttachmentValidator{ctrl: ctrl}
+	mock.recorder = &MockiAttachmentValidatorMockRecorder{mock}
+	return mock
+}
+
+// EXPECT returns an object that allows the caller to indicate expected use.
+func (m *MockiAttachmentValidator) EXPECT() *MockiAttachmentValidatorMockRecorder {
+	return m.recorder
+}
+
+// evalSymlinks mocks base method.
+func (m *MockiAttachmentValidator) evalSymlinks(path string) (string, error) {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "evalSymlinks", path)
+	ret0, _ := ret[0].(string)
+	ret1, _ := ret[1].(error)
+	return ret0, ret1
+}
+
+// evalSymlinks indicates an expected call of evalSymlinks.
+func (mr *MockiAttachmentValidatorMockRecorder) evalSymlinks(path interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "evalSymlinks", reflect.TypeOf((*MockiAttachmentValidator)(nil).evalSymlinks), path)
+}
+
+// readFile mocks base method.
+func (m *MockiAttachmentValidator) readFile(name string) ([]byte, error) {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "readFile", name)
+	ret0, _ := ret[0].([]byte)
+	ret1, _ := ret[1].(error)
+	return ret0, ret1
+}
+
+// readFile indicates an expected call of readFile.
+func (mr *MockiAttachmentValidatorMockRecorder) readFile(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "readFile", reflect.TypeOf((*MockiAttachmentValidator)(nil).readFile), name)
+}
+
+// MockMounter is a mock of Mounter interface.
+type MockMounter struct {
+	ctrl     *gomock.Controller
+	recorder *MockMounterMockRecorder
+}
+
+// MockMounterMockRecorder is the mock recorder for MockMounter.
+type MockMounterMockRecorder struct {
+	mock *MockMounter
+}
+
+// NewMockMounter creates a new mock instance.
+func NewMockMounter(ctrl *gomock.Controller) *MockMounter {
```

Comment:
As a side note, I'm personally more a fan of stubbing than mocking when possible since it's usually simpler (at the price of a bit more verbosity). For instance, our test attachment validator could look like this:

```go
type testAttachmentValidator struct {
  readFileFunc func(name string) ([]byte, error)
  evalSymlinksFunc func(path string) (string, error)
}

func (av *testAttachmentValidator) readFile(name string) ([]byte, error) {
  return av.readFileFunc(name)
}

func (av *testAttachmentValidator) evalSymlinks(path string) (string, error) {
  return av.evalSymlinksFunc(path)
}
```

This would seem sufficient as we do not need to verify that certain methods are being called -- it should conclude naturally from the fact that we have to rely on the interface and it being implemented appropriately.

Using a stub also means we don't need extra tooling and processes to update the mocks if the interface changes (which is something I'd otherwise ask to be documented in this PR).

Reply:
Sure. I've refactored it to use your struct. It's cleaner and one dependency/package less to maintain.